### ISSUE TITLE
No complaint for unusual doc comment for constant

### ISSUE THREAD

**Issue by kytrinyx** [golang/lint#166]
I was reviewing a bit of code someone wrote and saw the following:

```
// Export TestVersion
const TestVersion = 1
```

I would have expected a complain of the format _comment on exported const TestVersion should be of the form "TestVersion ..."_.

Have the guidelines on how to comment exported constants changed?

**Comment by dsymonds** [comment]
Nothing has changed. If that was in a test file it wouldn't be reported, though.

However, a bug might have snuck in at some point. We suppress the exact requirements for consts if the const is in a const block with a comment, like
```go
// Lots of constants
const (
  Foo = 1
  // missing something
  Bar = 2
  // watch out for this next one
  Baz = 3
)
```

Perhaps the change to do that meant that standalone consts don't ever get the form of their comment checked. That's a bug.

**Comment by kytrinyx** [comment]
The file is not a test file; it's in a small single-file package (not "main").

I went back in time, and it earlier versions are not complaining about this code example at all, so it doesn't look like this is a bug introduced when changing the recommendations for the const block.

I went and checked other code examples that I have, and I _am_ getting the expected complaint elsewhere if I put a different word before the name of the constant, so there seems to be a reason why this particular example isn't triggering the complaint.

**Comment by kytrinyx** [comment]
I'm trying to narrow down this issue, and am using a single-file package, thing.

```
./thing
└── thing.go
```

The source of `thing.go` is:

```
package thing

// hi
const Stuff = 1
```

The `lintValueSpecDoc()` function gets called as expected with the 

```
vs.Doc: (*ast.CommentGroup)(nil)
vs.Names[0]: Stuff
gd.Doc.List[0]: &ast.Comment{Slash:16, Text:"// hi"}
```

Because vs.Doc is nil at `lint.go:838`, the function returns before it gets to the bit where it can complain about the format of the comment.

I'm not familiar with golint or the ast package so I'm not sure where to go next with this. Is the nil Doc expected in the ValueSpec?

**Comment by dsymonds** [comment]
I found the problem. For standalone package-globals, the doc can appear on the ValueSpec or the GenDecl depending on whether the GenDecl has parens. I just have to reshuffle the logic slightly.

**Comment by kytrinyx** [comment]
Thank you!
